v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/cassandra/rowmapper/RowMapper.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.cassandra.rowmapper;
+
+import com.datastax.driver.core.Row;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.teleport.v2.source.reader.io.cassandra.mappings.CassandraMappingsProvider;
+import com.google.cloud.teleport.v2.source.reader.io.row.SourceRow;
+import com.google.cloud.teleport.v2.source.reader.io.schema.SourceSchemaReference;
+import com.google.cloud.teleport.v2.source.reader.io.schema.SourceTableSchema;
+import com.google.common.collect.ImmutableMap;
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.Schema;
+import org.apache.commons.collections4.Transformer;
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.UnknownKeyFor;
+
+@AutoValue
+public abstract class RowMapper implements Transformer<Row, SourceRow>, Serializable {
```

Comment:
`CassandraMappingsProvider` is accessed outside Cassandra package as a Unified Mapping Provider.

RowMapper will be using by `io.cassandra` package - It needs to be public as it will be accessed outside it's immediate package, but still it will be used only within the `io.cassandra` umbrella.